app/controllers/carts_controller.rb

```diff
@@ -7,4 +8,19 @@ def show
     @show_comments = true
   end
 
+  def index
+    @role = 'requester'
+    my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
+    if my_carts.empty?
+      @role = 'approver'
+      my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
+    end
```

Comment:
Sure, I'm fine with that. I'll add the comment.

On Fri, Jan 2, 2015 at 12:34 PM, Aidan Feldman <notifications@github.com>
wrote:

> In app/controllers/carts_controller.rb
> <https://github.com/18F/C2/pull/131#discussion-diff-22421945>:
>
> > @@ -7,4 +8,19 @@ def show
> >      @show_comments = true
> >    end
> >
> > +  def index
> > +    @role = 'requester'
> > +    my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
> > +    if my_carts.empty?
> > +      @role = 'approver'
> > +      my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
> > +    end
>
> Ohhh I see the issue.. the @role wouldn't be set. I think the logic makes
> more sense on a cart-by-cart basis:
>
> for each cart
>   show the cart summary
>   if user is an approver on the cart
>     show action buttons
>
> rather than one role for the overall page. If you agree, feel free to add
> an inline TODO and we can worry about it later. Otherwise, no worries.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/18F/C2/pull/131/files#r22421945>.
>